supervisor.go

```diff
@@ -0,0 +1,123 @@
+package streams
+
+// Supervisor represents a commit supervisor.
+//
+// The Supervisor performs a commit in a concurrently-safe manner.
+// There can only ever be 1 ongoing commit at any given time.
+type Supervisor interface {
+	// Perform a global commit sequence.
+	//
+	// If triggered by a Node, that node should be passed as an argument to Commit.
+	Commit(Processor) error
+}
+
+type sourceMetadata map[Source]Metadata
+
+// Merge merges metadata from metaitems.
+//
+// TODO: Incorporate merge strategies.
+func (m sourceMetadata) Merge(items []Metaitem) sourceMetadata {
+	for _, item := range items {
+		m[item.Source] = item.Metadata.Merge(m[item.Source])
+	}
+	return m
+}
+
+// Compile-type interface check.
+var _ Supervisor = (*supervisor)(nil)
+
+type supervisor struct {
+	store Metastore
+
+	pumps map[Processor]Pump
+
+	committerMeta []Metaitem
```

Comment:
Metadata is not carried between commits. Is this actually needed?

Reply:
It's a buffer for merging. After our discussion - will be removed anyway.